janus/server/api.py

```diff
@@ -21,8 +22,11 @@
 )
 def collect_qc(collect_request: CollectQCRequest = Body(...)) -> CollectQCResponse | JSONResponse:
     """Collect qc metrics for the external request."""
-    service = CollectQCService(collect_request)
     try:
+        workflow_service = get_workflow_service(collect_request.workflow)
```

Comment:
typehint

Should we create an empty and more of a general parent class for BalsamicCollectQCService, and use it as typehint here?

Reply:
Yes!

Reply:
If you want to, it could be an abstract class